### PR TITLE
Group generate egress key (#135)

### PR DESCRIPTION
**Pull request by speed47**

### CODE REVIEW

**Comment by speed47** [comment]
This implements #135

File: bin/plugin/group-owner/groupDelEgressKey [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+if (!OVH::Bastion::is_group_owner(account => $self, group => $shortGroup, superowner => 1)) {
+    osh_exit 'ERR_NOT_GROUP_OWNER', "You must be an owner to delete an egress group key";
+}
+
+$fnret = OVH::Bastion::get_group_keys(group => $group);
+
+my @matchingKeys = grep { $fnret->value->{'keys'}{$_}{'id'} eq $id } @{$fnret->value->{'sortedKeys'} || []};
+
+if (!@matchingKeys) {
+    osh_exit 'ERR_INVALID_PARAMETER', "Couldn't find any key with the ID you specified ($id) in group $shortGroup";
```
Comment (Alkorin) [inline bin/plugin/group-owner/groupDelEgressKey:54]:
```suggestion
    osh_exit 'ERR_INVALID_PARAMETER', "Couldn't find any key with the ID you specified ($id) in group $group";
```
?

File: bin/helper/osh-groupDelEgressKey [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+#<RIGHTSCHECK
+
+#>CODE
+$fnret = OVH::Bastion::get_group_keys(group => $group);
+$fnret or HEXIT($fnret);
+
+my @matchingKeys = grep { $fnret->value->{'keys'}{$_}{'id'} eq $id } @{$fnret->value->{'sortedKeys'} || []};
+
+if (!@matchingKeys) {
+    HEXIT('ERR_INVALID_PARAMETER', msg => "Couldn't find any key with the ID you specified ($id) in group $shortGroup");
```
Comment (Alkorin) [inline bin/helper/osh-groupDelEgressKey:85]:
```suggestion
    HEXIT('ERR_INVALID_PARAMETER', msg => "Couldn't find any key with the ID you specified ($id) in group $group");
```

File: bin/plugin/group-owner/groupModify [hunk 3]
```diff
@@ -49,6 +49,11 @@ $fnret or osh_exit $fnret;
 $group = $fnret->value->{'group'};
 my $shortGroup = $fnret->value->{'shortGroup'};
 
+$fnret = OVH::Bastion::is_group_owner(account => $self, group => $shortGroup, superowner => 1);
+if (!$fnret) {
+    osh_exit 'ERR_NOT_GROUP_OWNER', "You must be an owner to delete an egress group key";
```
Comment (Alkorin) [inline bin/plugin/group-owner/groupModify:54]: Wrong message ?

File: lib/perl/OVH/Bastion/Plugin/generateEgressKey.pm [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        }
+        $fnret = OVH::Bastion::is_valid_group_and_existing(group => $group, groupType => 'key');
+        $fnret or return $fnret;
+        my $keyhome    = $fnret->value->{'keyhome'};
+        my $shortGroup = $fnret->value->{'shortGroup'};
+        $group = $fnret->value->{'group'};
+
+        $fnret = OVH::Bastion::is_group_owner(group => $shortGroup, account => $self, superowner => 1, sudo => $sudo);
+        if (!$fnret) {
+            return R('ERR_NOT_GROUP_OWNER', msg => "Sorry, you're not an owner of group $shortGroup, which is needed to manage its egress keys ($fnret)");
```
Comment (Alkorin) [inline lib/perl/OVH/Bastion/Plugin/generateEgressKey.pm:105]:
```suggestion
            return R('ERR_NOT_GROUP_OWNER', msg => "Sorry, you're not an owner of group $group, which is needed to manage its egress keys ($fnret)");
```

File: lib/perl/OVH/Bastion/execute.inc [hunk 5]
```diff
@@ -67,20 +67,16 @@ sub execute {
     }
 =cut
 
-    #=cut only to debug tainted stuff
```
Comment (Alkorin) [inline lib/perl/OVH/Bastion/execute.inc:70]: Mmm ?

**Review by Alkorin — CHANGES_REQUESTED** [review]

**Review by Alkorin — APPROVED** [review]

**Review by deathiop — APPROVED** [review]

**Pull request merged**